src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -30,51 +30,51 @@ def call
                      else
                        "https://#{host}/api/v3/"
                      end
       {
         scm: 'github',
-        commit_sha: @payload.dig('pull_request', 'head', 'sha'),
-        pr_number: @payload['number'],
-        source_branch: @payload.dig('pull_request', 'head', 'ref'),
-        target_branch: @payload.dig('pull_request', 'base', 'ref'),
-        action: @payload['action'], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
-        source_repository_full_name: @payload.dig('pull_request', 'head', 'repo', 'full_name'),
-        target_repository_full_name: @payload.dig('pull_request', 'base', 'repo', 'full_name'),
+        commit_sha: @payload.dig(:pull_request, :head, :sha),
+        pr_number: @payload[:number],
+        source_branch: @payload.dig(:pull_request, :head, :ref),
+        target_branch: @payload.dig(:pull_request, :base, :ref),
+        action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
+        source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+        target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name),
         event: @event,
         api_endpoint: api_endpoint
-      }.with_indifferent_access
+      }.deep_symbolize_keys
     end
 
     def gitlab_extractor_payload
-      http_url = @payload.dig('project', 'http_url')
+      http_url = @payload.dig(:project, :http_url)
       uri = URI.parse(http_url)
       {
         scm: 'gitlab',
-        object_kind: @payload['object_kind'],
+        object_kind: @payload[:object_kind],
         http_url: http_url,
-        commit_sha: @payload.dig('object_attributes', 'last_commit', 'id'),
-        pr_number: @payload.dig('object_attributes', 'iid'),
-        source_branch: @payload.dig('object_attributes', 'source_branch'),
-        target_branch: @payload.dig('object_attributes', 'target_branch'),
-        action: @payload.dig('object_attributes', 'action'), # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
-        project_id: @payload.dig('object_attributes', 'source_project_id'),
-        path_with_namespace: @payload.dig('project', 'path_with_namespace'),
+        commit_sha: @payload.dig(:object_attributes, :last_commit, :id),
+        pr_number: @payload.dig(:object_attributes, :iid),
+        source_branch: @payload.dig(:object_attributes, :source_branch),
+        target_branch: @payload.dig(:object_attributes, :target_branch),
+        action: @payload.dig(:object_attributes, :action), # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
+        project_id: @payload.dig(:object_attributes, :source_project_id),
+        path_with_namespace: @payload.dig(:project, :path_with_namespace),
         event: @event,
         api_endpoint: "#{uri.scheme}://#{uri.host}"
-      }.with_indifferent_access
+      }.deep_symbolize_keys
```

Comment:
Same comment as for `github_extractor_payload`.

Here, we can also safely remove `deep_symbolize_keys`:
```suggestion
      }
```